File: solutions/python/high-scores/high_scores.py
"""High score utilities."""


class HighScores:
    """High score utility class."""
    def __init__(self, scores):
        """Initialize."""
        self.scores = scores

    def personal_top_three(self) -> list[int]:
        """Return the top three."""
        return sorted(self.scores, reverse=True)[:3]

    def personal_best(self) -> int:
        """Return the best score."""
        return max(self.scores)

    def latest(self) -> int:
        """Return the last score."""
        return self.scores[-1]

File: solutions/ruby/high-scores/high_scores.rb
class HighScores
  private

  attr_writer :scores

  def initialize(scores)
    self.scores = scores
  end

  public

  attr_reader :scores

  def personal_top_three()= scores.sort.reverse[0, 3]

  def latest()= scores[-1]

  def personal_best()= scores.max

  def latest_is_personal_best?()= latest == personal_best
end
